aws/resource_aws_msk_cluster.go

```diff
@@ -643,6 +634,40 @@ func resourceAwsMskClusterUpdate(d *schema.ResourceData, meta interface{}) error
 		}
 	}
 
+	if d.HasChanges("encryption_info", "client_authentication") {
+
+		input := &kafka.UpdateSecurityInput{
+			ClusterArn:     aws.String(d.Id()),
+			CurrentVersion: aws.String(d.Get("current_version").(string)),
+		}
+
+		if d.HasChange("client_authentication") {
```

Comment:
These checks are probably not necessary. I attempted to send only the changes applied as whilst the API is idempotent it still takes ~25 for an operation to complete even if not changes are made.

For example if the cluster is provisioned without a `client_authentication` block sasl is implicilty disabled, if we later add the configuration and explicitly set it disabled this will result still result in a cluster update

```hcl
  client_authentication {
    sasl {
      sasl= false
    }
  }
```